### DISCUSSION TITLE
Add --region option to select an exact screenshot region

### DISCUSSION THREAD

**Discussion by veracioux** [flameshot-org/flameshot#1974] [Ideas]
I propose a common solution to the following issues:
- #479
- #523
- #273

(The discussions in the issues became too complicated to follow)

**The problem in a nutshell**

You can't select the whole screen or a specific region without complicated and hacky scripts.

**Proposed solution**

Add a `--region` option to the `gui` and `screen` subcommands. This option can be used in the following ways:
- `--region FORMAT` (see the table below)
- `--region all` (select the entire screen, not available with the `screen` subcommand)
- `--region screen0`, `--region screen1`, etc. (select an entire screen by number, not available with the `screen` subcommand)

| Format | Pros | Cons |
| --- | --- | --- |
| X,Y,W,H | intuitive | non-standard |
| WxH+x+y | used by other CLI tools (xrandr, imagemagick) | may confuse new shell-users (minor IMO) |
| WxH,x,y | N/A | N/A |

\* *x and y are the coordinates of the top-left corner of the region*
\* *w and h are the width and height*

We would also modify the `--print-geometry` to use the format we agree upon (currently uses `W H X Y`). We can even make `--region` accept all of the formats.

:bangbang: Please vote for **one or more** of these formats in the comments below. Please vote even if you are indifferent. :bangbang:

**Other isues that may benefit from this:**
- #334
- #417
- #1509
- #1902

**What I'd like everyone to do**
- If you have other suggestions, post them as comments. We will discuss them, and if they can't be included into the existing proposal, I may ask you to open a new issue.
- Please vote for this post if you agree with the proposal.
- Vote for one of the suggested region formats
- If you prefer someone else's suggestion, vote for it

**Comment by veracioux** [comment]
Vote for the `WxH+x+y` format.

**Comment by veracioux** [comment]
Vote for the `X,Y,W,H` format.

**Comment by veracioux** [comment]
Vote for the `WxH,x,y` format.

**Comment by deragon** [comment]
Why settle for one format?  You could easily support both last formats.  It is obvious that any non digit is a separator.  Thus a regex like `(\d)+[^\d+](\d)+[^\d+](\d)+[^\d+](\d)+[^\d+]` could easily extract all the formats using the order W H x y.

BTW, I do not find the first format X,Y,W,H more intuitive than the others, order wise.

**Reply by veracioux** [reply]
> Why settle for one format? You could easily support both last formats.

Yeah, I did mention that in my post. Should have emphasized it more. We can support both formats with the `--region` option, but for `--print-geometry` we should settle with one.

Thanks for the regex.

**Reply by deragon** [reply]
Since WxH+x+y is used by other CLI tools (xrandr, imagemagick), it seams to have become a de-facto standard.  Go for it.
